propkatraj/propkatraj.py

```diff
@@ -2,22 +2,239 @@
+
+    To do this you first must
+
+    1. Ensure that any molecules that have been broken across periodic
+       bondaries have been made whole.
+    2. Only supply molecules that :program:`propka` will support.
+
+    To generate a timeseries of pKa predictions for a protein, first create
+    a :class:`PropkaTraj` object by supplying an MDAnalysis Universe or
+    AtomGroup::
+
+      import MDAnalysis as mda
+      # here we will use the MDAnalysis PSF and DCD example files
+      from MDAnalysisTests.datafiles import PSF, DCD
+      from propkatraj import PropkaTraj
+
+      u = mda.Universe(PSF, DCD)
+
+      pkatraj = PropkaTraj(u)
+
+    By default :class:`PropkaTraj` will select all protein residues (based on
+    standard protein residue naming) in the input ``atomgroup``, however this
+    can be changed by passing the ``select`` keyword. For example to limit the
+    input residues to residues numbered 1 through to 10::
+
+      pkatraj = PropkaTraj(u, select="resnum 1-10")
+
+    Alternatively, you can create an atomgroup beforehand and pass `None` to
+    ``select``::
+
+      ag = u.select_atoms('resnum 1-10")
+      pkatraj = PropkaTraj(ag, select=None)
+
+    Once the :class:`PropkaTraj` object has been created, we then use the
+    :meth:`run` to analyse the trajectory::
+
+      pkatraj.run()
+
+    In some cases, especially considering that calling propka can be time
+    consuming for large systems, you may not want to analyse every frame.
+    You can pass the ``start``, ``stop``, and ``step`` arguments to :meth:`run`
+    in order to alter the starting frame, final frame and frame sampling
+    frequency. Please note that frame numbers are zero formatted, and the
+    ``stop`` keyword is exclusive, as is the default in
+    :class:`MDAnalysis.AnalysisBase` objects. For example, in order to sample
+    every two frames from the first 10 frames::
+
+      pkatraj.run(start=0, stop=10, step=2)
+
+    You can also set the ``verbose`` keyword in order to have a nice progress
+    bar of the analysis progress::
+
+      pkatraj.run(verbose=True)
+
+    Once completed, the resulting timeseries of predicted pKas per residue is
+    made available as a :class:`pandas.DataFrame` under :attr`PropkaTraj`.pkas.
+    For example if one wanted to get a summary of the statistics of the
+    timeseries::
+
+      pkatraj.pkas.describe()
+
+    If one wanted to plot per residue boxplots of the timeseries data and save
+    it as a file `pKa-plot.png` (note: this requires the ``matplotlib`` and
+    ``seaborn`` packages which are not installed by default with
+    ``propkatraj``)::
+
+      import matplotlib.pyplot as plt
+      import seaborn as sns
+
+      fig = plt.figure(figsize=(28, 8))
+      ax = fig.add_subplot(1, 1, 1)
+      sns.boxplot(data=pkatraj.pkas, ax=ax)
+      ax.set_xlabel('residue number')
+      ax.set_ylabel(r'p$K_a$')
+      fig.savefig('pKa-plot.png')
+
+
+    .. versionadded:: 1.0.3
```

Comment:
Let's make this version 1.1.0 instead. This is a major addition but fully backwards compatible.